### DISCUSSION TITLE
Storylines Editor intranet path issues

### DISCUSSION THREAD

**Discussion by szczz** [ramp4-pcar4/storylines-editor#249] [General]
While testing the editor on the intranet we noticed that logos, charts, RAMP instances etc were not loading as expected when viewing them through the editor-preview endpoint. 

For example if we navigate to `{intranet-domain}/editor-preview/demo_test-test_demo` the charts don't render and the RAMP instance doesn't load. However, if we load this through the editor and press preview then ✨ all good.

We have a few console errors that indicate we're attempting to load these resources / ramp config from the wrong spot. These are located in the public folder, however the editor isn't using the /public path. Looking at the config for `demo_test-test_demo` we see the following path for the logo

![image](https://github.com/ramp4-pcar4/storylines-editor/assets/28012494/d495c2b9-db4f-416f-936b-dca4183e085d)

The console is grousin that it cannot get `{intranet-domain}/demo_test-test_demo/assets/en/worksonmymachine.png` which makes sense because the actual path is `{intranet-domain}/public/demo_test-test_demo/assets/en/worksonmymachine.png`. 

After buggin Yi Lei all afternoon, he found that the logo is being fetched from the storylines plugin in `introduction.vue`. 

We have some options:

1) Prefix this route with /public/ in the plugin (no bueno)
2) Update the code to prefix its src with /public/ if we're saving a new storylines on the intranet. However, this could be problematic as we need to support existing storylines.
3) Yi Lei suggested to construct the configfilestructure object when accessing the URL directly

Figured we should open this up for discussion and determine a path forward.

**Comment by yileifeng** [comment]
Not sure how complex constructing the `configFileStructure` object would be, but prefer this option over the other 2 solutions. Will bring consistent behavior as accessing the preview mode from the editor itself, since the missing `configFileStructure` prop is the only difference between these two methods of accessing preview.

**Reply by RyanCoulsonCA** [reply]
I also prefer this option. `metadata-editor.vue` already has code to fetch the files from the server and convert it into a `configFileStructure` object (with the help of the `configFileStructureHelper` function. See [here](https://github.com/ramp4-pcar4/storylines-editor/blob/main/src/components/editor/metadata-editor.vue#L389) for the code), so hopefully it shouldn't be too complex.

**Comment by szczz** [comment]
Constructing the configFileStructure was a success. Charts and RAMP instances are loading nicely now 🏆. However, we're still running into the logo path issue `{intranet-domain}/demo_test-test_demo/assets/en/worksonmymachine.png ` vs `{intranet-domain}/public/demo_test-test_demo/assets/en/worksonmymachine.png `. The logo didn't load when we entered the preview from /editor-main before the configFileStructure changes, but it did/does load when we supply the UUID from editor-metadata. Seems to be unrelated to configFileStructure, so I think we need to sprinkle in some of option 1 into this solution.
